app/src/main/java/com/jalasoft/routesapp/ui/auth/registerUser/view/RegisterUserFragment.kt

```diff
@@ -74,7 +151,55 @@ class RegisterUserFragment : Fragment() {
         val password = binding.etRegPassword.text.toString()
         val confirmPassword = binding.etRegConfirmPassword.text.toString()
 
-        viewModel.registerUserAuth(name, email, password, confirmPassword)
+        viewModel.verifyRegisterUserAuth(name, email, password, confirmPassword)
+    }
+
+    private fun signInGoogle() {
+        val signInIntent = googleSingInClient.signInIntent
+        launcher.launch(signInIntent)
+    }
+
+    private val launcher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(result.data)
+            handleGoogleResults(task)
+        }
+    }
+
+    private fun handleGoogleResults(task: Task<GoogleSignInAccount>) {
+        val account: GoogleSignInAccount? = task.result
+        if (account != null) {
+            val credential = GoogleAuthProvider.getCredential(account.idToken, null)
+            updateUI(account.displayName.toString(), account.email.toString(), UserTypeLogin.GOOGLE, credential)
+        }
+    }
+
+    private fun handleFacebookAccessToken(token: AccessToken) {
+        val credential = FacebookAuthProvider.getCredential(token.token)
+        val request = GraphRequest.newMeRequest(token) { obj, _ ->
```

Comment:
This call is the same on both fragments, shouldn't it be handled on a Manager / Repository?. Remember Single responsibility principle.

Reply:
Solved